app/assets/javascripts/discourse/models/user.js

```diff
@@ -840,6 +840,21 @@ const User = RestModel.extend({
       !secondFactorEnabled &&
       (enforce === "all" || (enforce === "staff" && staff))
     );
+  },
+
+  resolvedTimezone() {
```

Comment:
I'm unsure this should be in the model.

At the very least I think async event from models should return the promise.

Reply:
@jjaffeux I did it this way because IMO the result of the AJAX call does not matter, we just want to get the timezone or guess it and transparently save in the background. Where would you put this if not the model? Because the user model has the timezone on it and this is updating something on the user model.

I will defer to your experience here and return the promise if you think that will work better, just wanted to explain my reasoning.